hack/run-e2e-kind.sh

```diff
@@ -27,49 +27,58 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-export ROOT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/..
+export ROOT_DIR="$(dirname "$(dirname "$(readlink -fm "$0")")")"
 export LOG_LEVEL=3
-export CLEANUP_CLUSTER=${CLEANUP_CLUSTER:-1}
+export CLEANUP_CLUSTER=${CLEANUP_CLUSTER:-"true"}
 export CLUSTER_CONTEXT="--name test"
 # Using older image due to older version of kubernetes cluster"
-export IMAGE_NGINX="nginx:1.15.12"
-export IMAGE_ECHOSERVER="k8s.gcr.io/echoserver:1.4"
+export IMAGE_ECHOSERVER="kicbase/echo-server:1.0"
+export IMAGE_UBUNTULATEST="ubuntu:latest"
 export KIND_OPT=${KIND_OPT:=" --config ${ROOT_DIR}/hack/e2e-kind-config.yaml"}
 export KA_BIN=_output/bin
 export WAIT_TIME="20s"
 export IMAGE_REPOSITORY_MCAD="${1}"
 export IMAGE_TAG_MCAD="${2}"
 export MCAD_IMAGE_PULL_POLICY="${3-Always}"
 export IMAGE_MCAD="${IMAGE_REPOSITORY_MCAD}:${IMAGE_TAG_MCAD}"
+CLUSTER_STARTED="false"
+
+function update_test_host {
+  sudo apt-get update && sudo apt-get install -y apt-transport-https curl 
+  curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
+  echo "deb https://apt.kubernetes.io/ kubernetes-xenial main" | sudo tee -a /etc/apt/sources.list.d/kubernetes.list
+  sudo apt-get update
+  # Using older version due to older version of kubernetes cluster"
+  sudo apt-get install -y --allow-unauthenticated kubectl=1.17.0-00
+
+  # Download kind binary (0.6.1)
+  sudo curl -o /usr/local/bin/kind -L https://github.com/kubernetes-sigs/kind/releases/download/v0.11.0/kind-linux-amd64
+  sudo chmod +x /usr/local/bin/kind
+
+  # Installing helm3
+  curl -fsSL -o ${ROOT_DIR}/get_helm.sh https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3
+  chmod 700 ${ROOT_DIR}/get_helm.sh
+  ${ROOT_DIR}/get_helm.sh
+  sleep 10
+}
 
-sudo apt-get update && sudo apt-get install -y apt-transport-https
-curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
-echo "deb https://apt.kubernetes.io/ kubernetes-xenial main" | sudo tee -a /etc/apt/sources.list.d/kubernetes.list
-sudo apt-get update
-# Using older version due to older version of kubernetes cluster"
-sudo apt-get install -y  --allow-unauthenticated kubectl=1.17.0-00
-
-# Download kind binary (0.6.1)
-sudo curl -o /usr/local/bin/kind -L https://github.com/kubernetes-sigs/kind/releases/download/v0.11.0/kind-linux-amd64
-sudo chmod +x /usr/local/bin/kind
-
-# check if kind installed
+# check if pre-requizites are installed.
 function check-prerequisites {
   echo "checking prerequisites"
   which kind >/dev/null 2>&1
-  if [[ $? -ne 0 ]]; then
+  if [ $? -ne 0 ]; then
```

Comment:
(FMEP) can you provide some details on this update?

Reply:
The `$?` is an integer, I wanted to keep the usage of the test condition in line with [the manual](https://tldp.org/LDP/abs/html/comparison-ops.html)

Reply:
Got it, thanks for the clarification!